client/my-sites/subscribers/components/subscribers-page/subscribers-page-context.tsx

```diff
@@ -62,8 +64,15 @@ export const SubscribersPageProvider = ( {
 	const [ showAddSubscribersModal, setShowAddSubscribersModal ] = useState( false );
 
 	const [ debouncedSearchTerm ] = useDebounce( searchTerm, 300 );
-
-	const grandTotalQueryResult = useSubscribersQuery( { siteId } );
+	const { data: subscribersTotals = { total: 0 } } = useSubscribersTotalsQueries( siteId || 0 );
+	const totalSubscribers = subscribersTotals?.total ?? 0;
+
+	// 30000 is the limit of subscribers that can be fetched without breaking the endpoint. This is a temporal solution.
+	filterOption =
```

Comment:
Overriding a prop feels a bit dirty to me. Could we use a new `const` here?

Reply:
Line 71

Reply:
Sure!

Reply:
I meant to use a new const when overriding `filterOption`. To me it feels like your props object passed to `SubscribersPageProvider` should be immutable and probably shouldn't be changed within the body. 

Something like

```
const subscribersPageFilterOption = filterOption === SubscribersFilterBy.All && hasManySubscribers
			? SubscribersFilterBy.WPCOM
			: filterOption;
```

(or even use useMemo, although the cost of a recalculation is minimal here).

This is pure nitpicking, but IMHO it helps readability.